packages/adapter-auto/index.js

```diff
@@ -46,72 +43,78 @@ async function import_from_cwd(name) {
 	return import(url);
 }
 
-for (const candidate of adapters) {
-	if (candidate.test()) {
-		/** @type {{ default: () => import('@sveltejs/kit').Adapter }} */
-		let module;
-
-		try {
-			module = await import_from_cwd(candidate.module);
-		} catch (error) {
-			if (
-				error.code === 'ERR_MODULE_NOT_FOUND' &&
-				error.message.startsWith(`Cannot find package '${candidate.module}'`)
-			) {
-				const package_manager = detect_package_manager();
-				const command = commands[package_manager](candidate.module);
-
-				try {
-					console.log(`Installing ${candidate.module}...`);
-
-					execSync(command, {
-						stdio: 'inherit',
-						env: {
-							...process.env,
-							NODE_ENV: undefined
-						}
-					});
-
-					module = await import_from_cwd(candidate.module);
-
-					console.log(`Successfully installed ${candidate.module}.`);
-					console.warn(
-						`\nIf you plan on staying on this deployment platform, consider replacing @sveltejs/adapter-auto with ${candidate.module}. This will give you faster and more robust installs, and more control over deployment configuration.\n`
-					);
-				} catch (e) {
-					throw new Error(
-						`Could not install ${candidate.module}. Please install it yourself by adding it to your package.json's devDependencies and try building your project again.`
-					);
-				}
-			} else {
-				throw error;
+/** @typedef {import('@sveltejs/kit').Adapter} Adapter */
+
+/**
+ * @returns {Promise<Adapter | undefined>} The corresponding adapter for the current environment if found otherwise undefined
+ */
+async function get_adapter() {
```

Comment:
I'd make it clearer from the name that this is installing the adapter as `get` sounds side-effect free
```suggestion
async function install_adapter() {
```

Reply:
In some cases it _won't_ install anything though. We could make it `get_or_install_adapter` or whatever but i think it's clear enough from context